extended/src/main/java/io/kubernetes/client/extended/workqueue/ratelimiter/ItemFastSlowRateLimiter.java

```diff
@@ -0,0 +1,48 @@
+package io.kubernetes.client.extended.workqueue.ratelimiter;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * ItemFastSlowRateLimiter does a quick retry for a certain number of attempts, then a slow retry
+ * after that
+ */
+public class ItemFastSlowRateLimiter<T> implements RateLimiter<T> {
+
+  private Map<T, Integer> failures;
+
+  private Duration fastDelay;
+  private Duration slowDelay;
+  private int maxFastAttempts;
+
+  public ItemFastSlowRateLimiter(Duration fastDelay, Duration slowDelay, int maxFastAttempts) {
+    this.fastDelay = fastDelay;
+    this.slowDelay = slowDelay;
+    this.maxFastAttempts = maxFastAttempts;
+
+    failures = new HashMap<>();
+  }
+
+  @Override
+  public synchronized Duration when(T item) {
```

Comment:
same as above, not using synchronized method here.

Reply:
again `ConcurrentHashMap` and `AtomicInteger` will provide what you need.

Reply:
Replaced with AtomicLongMap.